tools/bbbsupervisor/bbbsupervisor.go

```diff
@@ -462,47 +466,75 @@ func handleMiddlewareNoBitcoindConnectivity(event watcherEvent, pState *supervis
 	}
 
-	if oldValue == newValue {
+	if wasActive == isActive {
 		return nil // no state change (do nothing)
-	} else if oldValue == -1 {
+	} else if wasActive == -1 {
 		// There is no prior state. Set `bitcoin_ibd` via bbbconfig.sh to true or false  (depending on the new state) just to be sure.
-		if newValue == 1 {
+		if isActive == 1 {
 			err := setBBBConfigValue("bitcoin_ibd", "true")
 			if err != nil {
 				return fmt.Errorf("Handling trigger %s: Initial set. Setting BBB config value to `true` failed: %v", event.trigger.String(), err)
 			}
 		} else {
-			err := setBBBConfigValue("bitcoin_ibd", "false")
+			// unset clearnet IBD redis key when the IBD is finished
+			err := unsetClearnetIDB()
+			if err != nil {
+				return fmt.Errorf("Handling trigger %s: %s", event.trigger.String(), err)
+			}
+			err = setBBBConfigValue("bitcoin_ibd", "false")
 			if err != nil {
 				return fmt.Errorf("Handling trigger %s: Initial set. Setting BBB config value `false` failed: %v", event.trigger.String(), err)
 			}
 		}
-		pState.prometheusLastStateIBD = newValue // set the initial value for the state
-	} else if oldValue == 1 && newValue == 0 { // IDB finished
-		err := setBBBConfigValue("bitcoin_ibd", "false")
+		pState.prometheusLastStateIBD = isActive // set the initial value for the state
+	} else if wasActive == 1 && isActive == 0 { // IBD finished
+		// unset clearnet IBD redis key when the IBD is finished
+		err := unsetClearnetIDB()
+		if err != nil {
+			return fmt.Errorf("Handling trigger %s: %s", event.trigger.String(), err)
+		}
+		err = setBBBConfigValue("bitcoin_ibd", "false")
 		if err != nil {
 			return fmt.Errorf("Handling trigger %s: setting BBB config value to `false` failed: %v", event.trigger.String(), err)
 		}
-		pState.prometheusLastStateIBD = newValue
-	} else if oldValue == 0 && newValue == 1 { // IDB (re)started
+		pState.prometheusLastStateIBD = isActive
+	} else if wasActive == 0 && isActive == 1 { // IBD (re)started
 		err := setBBBConfigValue("bitcoin_ibd", "true")
 		if err != nil {
 			return fmt.Errorf("Handling trigger %s: setting BBB config value to `true` failed: %v", event.trigger.String(), err)
 		}
-		pState.prometheusLastStateIBD = newValue
+		pState.prometheusLastStateIBD = isActive
 	}
 	return nil
 }
 
+// unsetClearnetIDB unsets (0 - download blocks over Tor) the ibdClearnetRedisKey if set.
+// The key can only be set back to 1 (download blocks over clearnet) via RPC.
+func unsetClearnetIDB() (err error) {
+	const ibdClearnetRedisKey string = "bitcoind:ibd-clearnet"
+	isIBDClearnet, err := getRedisInt(ibdClearnetRedisKey)
+	if err != nil {
+		return fmt.Errorf("getting redis key %s failed: %v", ibdClearnetRedisKey, err)
+	}
+	if isIBDClearnet == 1 {
+		log.Printf("IDB finished. Setting %s to %d.\n", ibdClearnetRedisKey, 0)
+		err := setBBBConfigValue("disable", "bitcoin_ibd_clearnet")
```

Comment:
IIUC, the function `setBBBConfigValue` would call the following script
```
bbb-config.sh set disable bitcoin_ibd_clearnet
```
This would be wrong, as it should be `bbb-config.sh disable bitcoin_ibd_clearnet`. Thinking of it, I could move the command from "enable/disable" section into "set/get". Then this line should be 
```
err := setBBBConfigValue("bitcoin_ibd_clearnet", "false")
```
Should I adjust the bash script?

Reply:
Yeah you are totally right.. I should check closer.. 

Ok. Let's move it from enable/disable to set/get. I'll fix it here.

Reply:
If changed, the middleware RPC for that added in 5b6029528e77552dff334555b22f14a131afb7bd needs to be updated as well.

Reply:
See draft pull-request https://github.com/digitalbitbox/bitbox-base/pull/190 to change bash script